src/baseclient.cc

```diff
@@ -283,7 +282,7 @@ minio::s3::BucketExistsResponse minio::s3::BaseClient::BucketExists(
   if (GetRegionResponse resp = GetRegion(args.bucket, args.region)) {
     region = resp.region;
   } else {
-    return (resp.code == "NoSuchBucket") ? false : resp;
+    return BucketExistsResponse(false);
```

Comment:
We lost actual response error for other than `NoSuchBucket`.
```suggestion
    return (resp.code == "NoSuchBucket") ? BucketExistsResponse(false) : resp;
```

Reply:
Good spot, the original semantics will shortly be restored.